### ISSUE TITLE
[Bug]  Internal Server Error: Failed to scrape URL. (Internal server error) - 400 Invalid schema for response_format 'websiteContent': In context=(), 'additionalProperties' is required to be supplied and to be false. 

### ISSUE THREAD

**Issue by RRua** [mendableai/firecrawl#883]
**Describe the Bug**
When using FirecrawlApp.app.scrape_url with a previously working valid schema, it raises the following error:

app.scrape_url(url, params={'formats': ['extract'], 'extract': {
                    'schema': schema,
                }})

requests.exceptions.HTTPError: Internal Server Error: Failed to scrape URL. (Internal server error) - 400 Invalid schema for response_format 'websiteContent': In context=(), 'additionalProperties' is required to be supplied and to be false. - No additional error details provided.

I've tried to place 'additionalProperties' at every level of the json schema, but still does not work

**To Reproduce**
Steps to reproduce the issue:
1. Configure the environment or settings with '...'
2. Run the command '...'
3. Observe the error or unexpected output at '...'
4. Log output/error message

**Expected Behavior**
A valid response

**Screenshots**
<img width="1422" alt="Captura de ecrã 2024-11-09, às 11 51 08" src="https://github.com/user-attachments/assets/ff245789-44a4-4b47-ab7c-73a1c4da4946">

**Environment (please complete the following information):**
- OS:  macOS
- Firecrawl Version:  1.0.4
- python: 3.9

**Logs**
If applicable, include detailed logs to help understand the problem.

**Additional Context**
Add any other context about the problem here, such as configuration specifics, network conditions, data volumes, etc.

**Comment by mogery** [comment]
Can you share the schema that provides this error please?

**Comment by RRua** [comment]
```
class MediaItem(BaseModel):
    link: str
    alt: Optional[str]


class ExtractSchema(BaseModel):
    title: str
    description: Optional[str]
    price: Optional[int]
    currency: Optional[str]
    property_type: Optional[str]
    publisher: Optional[str]
    sq_meters: Optional[int]
    condition: Optional[str]
    created_at: Optional[str]
    updated_at: Optional[str]
    bedrooms: Optional[int]
    bathrooms: Optional[int]
    garages: Optional[int]
    floors: Optional[int]
    basement: Optional[int]
    garden: Optional[int]
    terrace: Optional[int]
    balcony: Optional[int]
    parking: Optional[int]
    swimming_pool: Optional[int]
    location: Optional[str]
    kitchen_type: Optional[str]
    heating_type: Optional[str]
    air_conditioning: Optional[bool]
    flooring_material: Optional[str]
    energy_rating: Optional[str]
    parish: Optional[str]
    images: Optional[List[MediaItem]]
    latitude: Optional[float]
    longitude: Optional[float]
    count_photos: Optional[int]

    @classmethod
    def model_json_schema_with_restrictions(cls) -> Dict[str, Any]:
        schema = cls.model_json_schema()
        schema['additionalProperties'] = False

        def set_additional_properties_false(sub_schema):
            if isinstance(sub_schema, dict):
                sub_schema['additionalProperties'] = False
                for key, value in sub_schema.get('properties', {}).items():
                    if value.get('type') == 'object':
                        set_additional_properties_false(value)

        set_additional_properties_false(schema)
        return schema
        
class X:
     .... 
      def scrape_with_llm_model(self, url: str, schema: dict, retries: int = 3, delay: int = 4) -> Dict[str, Any]:
        for attempt in range(retries):
            try:
                data = self.app.scrape_url(url, params={'formats': ['extract'], 'extract': {
                    'schema': schema,
                }})
                
                return data["extract"]
            except Exception as e:
                log(f"Attempt {attempt + 1} failed: {e}")
                
                if attempt < retries - 1:
                    log(f"Retrying in {delay} seconds...")
                    time.sleep(delay)
                else:
                    log("Max retries reached. Exiting.")
                    return {}

      def method_x():
           schema = ExtractSchema.model_json_schema_with_restrictions()  # Even tried with this
           property_content = self.scrape_with_llm_model(url, schema)
```

**Comment by yupingsong-anylink-io** [comment]
I also encountered the same problem.
My schema used to work properly, but now I suddenly encounter the error `'additional properties' is required`

**Comment by mogery** [comment]
Some more context for this issue: We have switched over to OpenAI's new Structured Outputs API to provide extract results with a better success rate, but they have stricter schema restrictions than we used to. We are working on applying a transformation to schemas to fix this issue. Thank you for reporting!

**Comment by mogery** [comment]
This has been resolved by e97864b, should work in prod now.
